votor/src/event_handler.rs

```diff
@@ -527,7 +527,15 @@ impl EventHandler {
         if *my_pubkey != new_pubkey || vctx.vote_history.node_pubkey != new_pubkey {
             let my_old_pubkey = vctx.vote_history.node_pubkey;
             *my_pubkey = new_pubkey;
-            vctx.vote_history = VoteHistory::restore(ctx.vote_history_storage.as_ref(), my_pubkey)?;
+            if let Ok(vote_history) =
+                VoteHistory::restore(ctx.vote_history_storage.as_ref(), my_pubkey)
+            {
+                warn!("Loaded vote history for {my_pubkey}: {vote_history:?}");
+                vctx.vote_history = vote_history;
+            } else {
+                warn!("No vote history found for {my_pubkey}, starting fresh");
```

Comment:
We shouldn't start fresh when calling set-identity as this could result in slashable votes. Unlike TowerBFT we don't have any information on chain to reconstruct our vote history - we should exit and the operator can fix it.

Reply:
Hmm, you mean, we should panic here, or return an error?

Reply:
Just what the code was doing before, returning an error so the validator can shutdown.

As an improvement we could consider keeping the validator running with the old identity but indicating to the user that set-identity failed - but it needs to be extremely clear in the CLI so that they don't accidentally set-identity on the other box and end up with 2 validators with the same identity.

Reply:
The file should be keyed by identity, e.g. in tower we have:
```
tower-1_9-{$IDENTITY}.bin
```
so on startup  of a hot spare, we populate a vote history file. Before set-identity is called the operator copies the vote history file of the staked identity. When set-identity is specified the validator should load the new vote history file and continue populating that one.

This is all untested at the moment, so you might have found a legitimate bug if this workflow isn't working

Reply:
Hmm okay, I think after manually creating a vote history storage file every time  before I switch identity I was able to make the tests pass.

Reply:
Awesome! I think for a future follow up we can expose a `--require-vote-history` flag in the CLI similar to `--require-tower` and make it evident in that not using the flag when switching to a *staked* identity can cause us to send slashable votes.

If we do that, i'd feel comfortable relaxing the check here.